### PR TITLE
Wagtail 6.0

### PR DESCRIPTION
**Pull request by katdom13**
[Support Ticket](https://torchbox.monday.com/boards/1124794299/pulses/1396934942)

### Added
- Add support for Wagtail 6.0

### Changed
- Implement stimulus approach to GoogleMapsField, GeocoderField, and LeafletField

### Removed
- Drop support for Django 4.1

### CODE REVIEW

**Review by nickmoreton — COMMENTED** [review]
Hi @katdom13 All looks OK to me.

But I wasn't able to use the address search anywhere. Did you get that part working? I can see the maps using my own maps key but may be the address search works with a different key.

**Comment by katdom13** [comment]
> Hi @katdom13 All looks OK to me.
> 
> But I wasn't able to use the address search anywhere. Did you get that part working? I can see the maps using my own maps key but may be the address search works with a different key.

Hi @nickmoreton ,
Could you let me know which page is that?
Thanks!

**Comment by katdom13** [comment]
> Hi @katdom13 All looks OK to me.
> 
> But I wasn't able to use the address search anywhere. Did you get that part working? I can see the maps using my own maps key but may be the address search works with a different key.

I think I see what you mean. This package supports searching from the following:
1. Google Geocoding - I noticed that this has type suggestions when I try to search.
2. Nominatim - This one doesn't, but the map should respond if I provide a complete address
3. Mapbox - This one doesn't work for me.

I've been testing this on `main` (without the changes) as well as the current work branch but there's no significant difference.

Thanks!

**Review by nickmoreton — APPROVED** [review]
Hi @katdom13 I think this is good. Nice work on the javascript 😄 
While I could see preview errors I just think it's because the testing site doesn't have templates mostly. But it's possible that this package was conceived before previews had there own panel.
I was able to add the correct services to my test api key and it took little while but address search did work in the end.

File: wagtailgeowidget/widgets.py [hunk 1]
```diff
@@ -39,153 +40,627 @@
...
+            if self.value_data and isinstance(self.value_data, str):
+                result = geosgeometry_str_to_struct(self.value_data)
+                if result:
+                    data["defaultLocation"] = {
+                        "lat": result["y"],
+                        "lng": result["x"],
+                    }
 
-            google_maps_apikey = callback()
+            if self.value_data and isinstance(self.value_data, Point):
```
Comment (nickmoreton) [inline wagtailgeowidget/widgets.py:85]: Just adding a note of an error I saw ion a client site:
```
File "/venv/lib/python3.8/site-packages/wagtailgeowidget/widgets.py", line 85, in build_attrs
    if self.value_data and isinstance(self.value_data, Point):
TypeError: isinstance() arg 2 must be a type or tuple of types
```

**Pull request merged**